In [1]:
import pandas as pd
from datetime import datetime as dt

In [8]:
#Read csv file
stations_df = pd.read_csv('Resources/hawaii_measurements.csv')
stations_df.head(10)

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73
5  USC00519397  2010-01-07  0.06    70
6  USC00519397  2010-01-08  0.00    64
7  USC00519397  2010-01-09  0.00    68
8  USC00519397  2010-01-10  0.00    73
9  USC00519397  2010-01-11  0.01    64

In [9]:
#Change date format to datetime
stations_df.date = pd.to_datetime(stations_df.date, infer_datetime_format=True)

In [10]:
#Set Date as the DF index
stations_df = stations_df.set_index(stations_df['date'])

In [6]:
#Remove date column, avoiding repeting columns
stations_df = stations_df.drop(columns='date')

In [7]:
stations_df.head(10)

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73
2010-01-07  USC00519397  0.06    70
2010-01-08  USC00519397  0.00    64
2010-01-09  USC00519397  0.00    68
2010-01-10  USC00519397  0.00    73
2010-01-11  USC00519397  0.01    64

## Comparing June and December data across years

In [12]:
from scipy import stats

In [13]:
# Filtering data
june = stations_df[stations_df.index.month == 6]
december = stations_df[stations_df.index.month == 12]

In [14]:
june.mean()

<ipython-input-14-6a3cd6125d74>:1: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  june.mean()


prcp     0.136360
tobs    74.944118
dtype: float64

In [15]:
december.mean()

<ipython-input-15-73accfa48a6e>:1: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  december.mean()


prcp     0.216819
tobs    71.041529
dtype: float64

In [18]:
#Temperature data
june_temp = june.tobs
december_temp = december.tobs

In [19]:
stats.ttest_ind(june_temp, december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

Analyzing all the available yearly data across all stations we got that the mean temperature in June was 74.944118 and the mean for December was 71.041529, this is a difference of 3.9 degrees. I did an unpaired ttest which result on a low pvalue = 3.90251, the difference between de months it is considered significant.